adapter/internal/sourcewatcher/sourcewatcher.go

```diff
@@ -0,0 +1,365 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package sourcewatcher
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/wso2/product-microgateway/adapter/config"
+	"github.com/wso2/product-microgateway/adapter/internal/api"
+	xds "github.com/wso2/product-microgateway/adapter/internal/discovery/xds"
+	"github.com/wso2/product-microgateway/adapter/internal/loggers"
+	"github.com/wso2/product-microgateway/adapter/internal/oasparser/model"
+	"github.com/wso2/product-microgateway/adapter/pkg/auth"
+	"github.com/wso2/product-microgateway/adapter/pkg/logging"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+const (
+	apisArtifactDir string = "apis"
+	zipExt          string = ".zip"
+)
+
+var artifactsMap map[string]model.ProjectAPI
```

Comment:
Do we always have to hold on to this in memory? As in, can't we load this from the local mounted storage when ever we see a change in the remote repo?

Reply:
The artifacts map was mainly added to delete the APIs currently deployed but not present in the repository after the update.